gCoffee/app/src/main/java/com/mborowiec/gcoffee/Order.java

```diff
@@ -98,4 +107,12 @@ private void makeOrder(String coffee) {
         Toast.makeText(this, "Ordered: " + coffee,
                 Toast.LENGTH_SHORT).show();
     }
+
+    private void handleDeepLinks(Uri data) {
+        String coffee;
```

Comment:
Did you consider to declare and initialize the variable in a single line, e.g., with

String coffee = data.getQuery...

?